### PR TITLE
M365 ans IAM script updates

### PR DESCRIPTION
**Pull request by Himalayk**
M365 and IAM agent script refinements.

### CODE REVIEW

File: scripts/Upgrade-AzureIAM-DataCollector.ps1 [hunk 1]
```diff
@@ -0,0 +1,213 @@
+<#
+.SYNOPSIS
+    Script to upgrade Cloudneeti IAM data collector automation account
```
Comment (rahulkhengare) [inline scripts/Upgrade-AzureIAM-DataCollector.ps1:3]: Script to upgrade Cloudneeti IAM data collector's automation account

File: scripts/Upgrade-AzureIAM-DataCollector.ps1 [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+.DESCRIPTION
+    This script will upgrade runbook from IAM data collector(Automation Account) and deprecate older version on runbook.
+.EXAMPLE
+    Commands to run this script.
+     Upload script to Azure CloudShell and execute below command:-
+    .\Upgrade-AzureIAM-DataCollector.ps1.ps1
```
Comment (rahulkhengare) [inline scripts/Upgrade-AzureIAM-DataCollector.ps1:9]: Remove double ps1

File: scripts/Upgrade-M365DataCollector.ps1 [hunk 3]
```diff
@@ -96,6 +107,12 @@ $ContianerName = "m365-datacollection-script"
 $RunbookScriptName = "$ScriptPrefix-$DataCollectorVersion.ps1"
 $RunbookName = "$ScriptPrefix-$DataCollectorVersion"
 $path = "./runbooks"
+$SharePointDomainURL = "https://$SharePointDomain"
```
Comment (rahulkhengare) [inline scripts/Upgrade-M365DataCollector.ps1:110]: convert SharepointDomainURL to SharePointAdminCenterURL as SharePoint domain URL is misleading

File: scripts/Provision-M365DataCollector.ps1 [hunk 4]
```diff
@@ -149,17 +149,27 @@ param
...
 
+    # Office Domain URL
+    [ValidateScript( {$_ -notmatch 'https://+' -and $_ -notmatch 'http://+'})]
+    [Parameter(Mandatory = $False,
+        HelpMessage = "Share Point Domain",
+        Position = 11
+    )]
+    [ValidateNotNullOrEmpty()]
+    [string]
+    $SharePointDomain = $(Read-Host -prompt "Enter Share Point Domain"),
```
Comment (rahulkhengare) [inline scripts/Provision-M365DataCollector.ps1:167]: convert **SharepointDomainURL** to **SharePointAdminCenterURL** as SharePoint domain URL is misleading

File: scripts/Provision-M365DataCollector.ps1 [hunk 5]
```diff
@@ -279,13 +291,13 @@ $RequiredModules = @"
             "Product": "SharePoint",
             "Name": "Microsoft.Online.SharePoint.PowerShell",
             "ContentUrl" : "https://www.powershellgallery.com/api/v2/package/Microsoft.Online.SharePoint.PowerShell/16.0.8414.1200",
-            "Version" : "16.0.8414.1200"
+            "Version" : "16.0.8414.1200"
```
Comment (rahulkhengare) [inline scripts/Provision-M365DataCollector.ps1:294]: remove spaces

File: scripts/Upgrade-AzureIAM-DataCollector.ps1 (hunk 1 above)
Comment (prateikw) [reply scripts/Upgrade-AzureIAM-DataCollector.ps1]: done

File: scripts/Upgrade-AzureIAM-DataCollector.ps1 (hunk 2 above)
Comment (prateikw) [reply scripts/Upgrade-AzureIAM-DataCollector.ps1]: done

File: scripts/Upgrade-M365DataCollector.ps1 (hunk 3 above)
Comment (prateikw) [reply scripts/Upgrade-M365DataCollector.ps1]: done

File: scripts/Provision-M365DataCollector.ps1 (hunk 4 above)
Comment (prateikw) [reply scripts/Provision-M365DataCollector.ps1]: done

File: scripts/Provision-M365DataCollector.ps1 (hunk 5 above)
Comment (prateikw) [reply scripts/Provision-M365DataCollector.ps1]: done

File: scripts/Provision-M365DataCollector.ps1 [hunk 6]
```diff
@@ -26,31 +26,41 @@
...
         - Cloudneeti Account Id
+        - Cloudneeti API Key
         - Cloudneeti Environment
+        - Cloudneeti Application Id
+        - Cloudneeti Application Secret
         - Cloudneeti Office 365 Data Collector Artifacts Storage Name
         - Cloudneeti Office 365 Data Collector Artifacts Storage Access Key
         - Cloudneeti Office 365 Data Collector Version
         - Office 365 Domain Name
+        - SharePoint Admin Center Domain
```
Comment (rahulkhengare) [inline scripts/Provision-M365DataCollector.ps1:37]: Sharepoint Admin Center URL

File: scripts/Provision-M365DataCollector.ps1 [hunk 7]
```diff
@@ -149,17 +159,27 @@ param
...
 
+    # Office Domain URL
+    [ValidateScript( {$_ -notmatch 'https://+' -and $_ -notmatch 'http://+'})]
+    [Parameter(Mandatory = $False,
+        HelpMessage = "Share Point Domain",
+        Position = 11
+    )]
+    [ValidateNotNullOrEmpty()]
+    [string]
+    $SharePointAdminCenterDomain = $(Read-Host -prompt "Enter SharePoint admin center domain name"),
```
Comment (rahulkhengare) [inline scripts/Provision-M365DataCollector.ps1:177]: SharePointAdminCenterURL

Help message correction

File: scripts/Provision-M365DataCollector.ps1 [hunk 8]
```diff
@@ -149,17 +159,27 @@ param
...
-        HelpMessage = "Office 365 Domain Name: ",
+        HelpMessage = "Office 365 Domain Name",
         Position = 10
     )]
     [ValidateNotNullOrEmpty()]
     [string]
     $OfficeDomain = $(Read-Host -prompt "Enter Office 365 Domain Name"),
 
+    # Office Domain URL
+    [ValidateScript( {$_ -notmatch 'https://+' -and $_ -notmatch 'http://+'})]
```
Comment (rahulkhengare) [inline scripts/Provision-M365DataCollector.ps1:170]: verify validation script

File: scripts/Provision-M365DataCollector.ps1 (hunk 6 above)
Comment (prateikw) [reply scripts/Provision-M365DataCollector.ps1]: done

File: scripts/Provision-M365DataCollector.ps1 (hunk 7 above)
Comment (prateikw) [reply scripts/Provision-M365DataCollector.ps1]: done

File: scripts/Provision-M365DataCollector.ps1 (hunk 8 above)
Comment (prateikw) [reply scripts/Provision-M365DataCollector.ps1]: done

**Pull request merged**